OpenRA.Game/Actor.cs

```diff
@@ -148,9 +175,32 @@ internal void Created()
 		{
 			created = true;
 
+			// Make sure traits are usable for condition notifiers
 			foreach (var t in TraitsImplementing<INotifyCreated>())
 				t.Created(this);
 
+			var allObserverNotifiers = new HashSet<VariableObserverNotifier>();
+			foreach (var provider in TraitsImplementing<IObservesVariables>())
+			{
+				foreach (var variableUser in provider.GetVariableObservers())
+				{
+					allObserverNotifiers.Add(variableUser.Notifier);
+					foreach (var variable in variableUser.Variables)
+					{
+						var cs = conditionStates.GetOrAdd(variable);
+						cs.Notifiers.Add(variableUser.Notifier);
+						conditionCache[variable] = 0;
+					}
+				}
+			}
+
```

Comment:
What happened to this code that used to live between these two in `ConditionManager.Created`?
```csharp
			// Enable any conditions granted during trait setup
			foreach (var kv in tokens)
			{
				ConditionState conditionState;
				if (!state.TryGetValue(kv.Value, out conditionState))
					continue;

				conditionState.Tokens.Add(kv.Key);
				conditionCache[kv.Value] = conditionState.Tokens.Count;
			}
```

Reply:
They are granted right away, but the notifications are postponed.

Reply:
I pushed a couple of fixes too.